test_rclcpp/test/test_spin.cpp

```diff
@@ -33,11 +33,23 @@
 
 using namespace std::chrono_literals;
 
+class CLASSNAME (test_spin, RMW_IMPLEMENTATION) : public ::testing::Test {
+public:
+  void SetUp() {
+    rclcpp::init(0, nullptr);
+  }
+
+  void TearDownTestCase() {
+    if (rclcpp::ok()) {
```

Comment:
For most of the other implementations, you just unconditionally call `rclcpp::shutdown`.  I guess this is a little better in that we won't try to `shutdown` if `init` failed somehow.  But I think that is pretty unlikely, so I'd just drop the `rclcpp::ok` check here.  Either way, I think we should be consistent among these tests.

Reply:
In this case, the test `spin_until_future_complete_interrupted` calls shutdown to interrupt an executor. So shutdown may or may not have been called. I think for most other files it should remain without the check so that we'll notice if there is an issue somewhere.